e2e-tests/main_test.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func runCommandWithOutput(cmd *exec.Cmd) (string, error) {
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if err != nil {
+		return "", fmt.Errorf("error running command: %v\nstdout: %s\nstderr: %s", err, stdout.String(), stderr.String())
+	}
+
+	return stdout.String(), nil
+}
+
+// From the output of the AbciInfo command, extract the latest block height.
+// The json bytes should look e.g. like this:
+// {"jsonrpc":"2.0","id":1,"result":{"response":{"data":"interchain-security-p","last_block_height":"2566","last_block_app_hash":"R4Q3Si7+t7TIidl2oTHcQRDNEz+lP0IDWhU5OI89psg="}}}
+func ExtractHeightFromInfo(jsonBytes []byte) (int, error) {
+	// Use a generic map to represent the JSON structure
+	var data map[string]interface{}
+
+	if err := json.Unmarshal(jsonBytes, &data); err != nil {
+		return 0, fmt.Errorf("Failed to unmarshal JSON %s \n error was %v", string(jsonBytes), err)
```

Comment:
Could 0 be a valid blockchain height or do chains start at 1? If 0 is valid, then maybe return -1.

Reply:
True, yes that could be problematic in the future, so I'm returning -1 now